src/Tests/VCEL.Test/FunctionTests.cs

```diff
@@ -1,384 +1,515 @@
+    [TestCase("substring('test_substring', 0, 0)", "")]
+    [TestCase("substring('test_substring', 0, 100)", null)]
+    [TestCase("substring('test_substring', 100, 100)", null)]
+    [TestCase("substring('test_substring', -1, 100)", null)]
+    [TestCase("substring('test_substring', -1, -1)", null)]
+    [TestCase("split('test_split', '_')", new[] { "test", "split" })]
+    [TestCase("split('test_split', null)", new string[0])]
+    [TestCase("split(null, '_')", new string[0])]
+    [TestCase("split(null, null)", new string[0])]
+    [TestCase("replace('test_replace', '_replace', '')", "test")]
+    [TestCase("replace('test_replace', '_replace', '_test')", "test_test")]
+    [TestCase("replace(null, '_replace', '_test')", null)]
+    [TestCase("replace('test_replace', null, '_test')", "test_replace")]
+    [TestCase("replace('test_replace', '_replace', null)", "test")]
+    [TestCase("replace(null, null, null)", null)]
+    [TestCase("trim(' test_trim ')", "test_trim")]
+    [TestCase("trim(null)", null)]
+    [TestCase("trim('')", "")]
+    [TestCase("length('test_length')", 11)]
+    [TestCase("length(null)", null)]
+    [TestCase("length('')", 0)]
+    [TestCase("contains('test_contains', 'test')", true)]
+    [TestCase("contains('test_contains', 'test1')", false)]
+    [TestCase("contains(null, 'test1')", false)]
+    [TestCase("startsWith('test_startswith', 'test')", true)]
+    [TestCase("startsWith('test_startswith', 'test1')", false)]
+    [TestCase("startsWith(null, 'test1')", false)]
+    [TestCase("endsWith('test_endswith', 'endswith')", true)]
+    [TestCase("endsWith('test_endswith', 'endswith1')", false)]
+    [TestCase("endsWith('test_endswith', 'test_endswith')", true)]
+    [TestCase("endsWith('test_endswith', 'test')", false)]
+    [TestCase("endsWith(null, 'endswith1')", false)]
+    [TestCase("indexOf('test_indexof', 'index')", 5)]
+    [TestCase("indexOf('test_indexof', 'index1')", null)]
+    [TestCase("indexOf(null, 'index1')", null)]
+    [TestCase("lastIndexOf('test_lastindexof', 'last')", 5)]
+    [TestCase("lastIndexOf('test_lastindexof', 'last1')", null)]
+    [TestCase("lastIndexOf('test_lastindexoflast', 'last')", 16)]
+    [TestCase("lastIndexOf(null, 'last1')", null)]
+    [TestCase("reverse('test_reverse')", "esrever_tset")]
+    [TestCase("reverse(null)", null)]
+    [TestCase("reverse('')", "")]
+    [TestCase("replace('test_replace', '_replace', '')", "test")]
+    [TestCase("replace('test_replace', '_replace', '_test')", "test_test")]
+    [TestCase("replace(null, '_replace', '_test')", null)]
+    [TestCase("replace('test_replace', null, '_test')", "test_replace")]
+    [TestCase("replace('test_replace', '_replace', null)", "test")]
+    [TestCase("replace(null, null, null)", null)]
+    [TestCase("get('test_get', 0)", "t")]
+    [TestCase("get('test_get', 1)", "e")]
+    [TestCase("get('test_get', 6)", "e")]
+    [TestCase("get('test_get', -1)", "t")]
+    [TestCase("get('test_get', -2)", "e")]
```

Comment:
These negative end lookups would be nice for the first positional arg of `substring` as well

Reply:
I'll leave as is for now, but agree would be a good change to make

Reply:
Agree, nice to have